### ISSUE TITLE
can't load channels, only on IPTV SIMPLE CLIENT in kodi. other apps working fine.

### ISSUE THREAD

**Issue by avsha114** [kodi-pvr/pvr.iptvsimple#454]
Hi!
I am trying to load channels but they won't load. tried both on windows and android.
the log shows:
2020-11-30 14:22:19.662 T:7240   DEBUG: CurlFile::Open(0x1a7377f98f0) http://m3u4u.com/XXXXXXXX.m3u
2020-11-30 14:22:19.672 T:17884   DEBUG: CAddonMgr::GetAvailableUpdates took 13 ms
2020-11-30 14:22:19.672 T:17884   DEBUG: Thread waiting 17884 terminating
2020-11-30 14:22:19.674 T:15584   DEBUG: Thread BackgroundLoader start, auto delete: false
2020-11-30 14:22:19.675 T:15584   DEBUG: Thread BackgroundLoader 15584 terminating
2020-11-30 14:22:19.873 T:7240   ERROR: CCurlFile::FillBuffer - Failed: HTTP returned error 403
2020-11-30 14:22:19.873 T:7240   ERROR: CCurlFile::Open failed with code 403 for http://m3u4u.com/XXXXXXXXXX.m3u:
2020-11-30 14:22:19.873 T:7240   ERROR: XFILE::CFileCache::Open - failed to open source <http://m3u4u.com/XXXXXXX.m3u>
2020-11-30 14:22:19.873 T:7240   ERROR: AddOnLog: PVR IPTV Simple Client: Unable to load playlist file 'http://m3u4u.com/XXXXXXXX.m3u':  file is missing or empty.

of course I replaced my real address for X's.
I tried on every app I could think (VLC, tivimate, iptv smarters...) and the link is working great on all of them. even if I paste the link in my browser it downloads it without any problem. it's a small file actually. the problem happens only on iptv simple client for kodi.

when I try to download the m3u file and open local m3u in my client it works!! but without grouping.
only the remote m3u link doesn't work.

any ideas?

EDIT:
I found something about mimetype but this is a remote URL. is there a way to add the mimetype to the URL options?
also, I tried both HLS and TS streams and hey both don't work on kodi while working on every other app.

**Comment by phunkyfish** [comment]
It's either something to do with http2: https://github.com/kodi-pvr/pvr.iptvsimple/issues/294

Or your provider may not accept the kodi user agent and you need to set a custom one. The README in the github repo should explain how to do that.

**Comment by avsha114** [comment]
thanks.
but it didn't work for me.
I even tried to install kodi 19 beta from scratch and created a new advancedsettings.xml file with:
```
<advancedsettings>
<network>
<disablehttp2>true</disablehttp2>
</network>
</advancedsettings>
```

or '1' instead if 'true'.
nothing worked.
still getting the message:

2020-12-01 01:57:54.572 T:17448   ERROR <general>: CCurlFile::FillBuffer - Failed: HTTP returned error 403
2020-12-01 01:57:54.572 T:17448   ERROR <general>: CCurlFile::Open failed with code 403 for http://m3u4u.com/XXXXX.m3u:
                                                   
2020-12-01 01:57:54.572 T:17448   ERROR <general>: XFILE::CFileCache::Open - failed to open source <http://m3uXXXXX.m3u>
2020-12-01 01:57:54.572 T:17448   ERROR <general>: AddOnLog: pvr.iptvsimple: pvr.iptvsimple - iptvsimple::PlaylistLoader::LoadPlayList - Unable to load playlist cache file 'http://m3u4uXXXXX.m3u':  file is missing or empty.

i guess i'll use another app :/
so frustrating...

**Comment by phunkyfish** [comment]
Did you try another user agent?

**Comment by phunkyfish** [comment]
You can try adding a `#EXTVLCOPT:http-user-agent=vlc` before the M3U entry or or appending `|user-agent=vlc` to the URL to try it.

**Comment by avsha114** [comment]
> You can try adding a `#EXTVLCOPT:http-user-agent=vlc` before the M3U entry or or appending `|user-agent=vlc` to the URL to try it.

I'm not sure if I can do this. I can't edit the m3u obviously because it's coming from the server.
and can you give me an example on how to URL should look? I am doing this through m3u4u.com which is a website that let's you edit the m3u and redownload it. so I can't really use any URL options I think.
I tried writing `vlc` in the user-agent field in IPTV simple client settings on kodi 19. no luck there.

**Comment by phunkyfish** [comment]
In kodi 19 there is an advanced setting to set this in iptvsimple settings. Try that.

EDIT: NVM, just read the end of your reply.

**Comment by demartini** [comment]
Hey @avsha114, I had the same problem, after almost giving up I found the answer on the m3u4u.com [wiki](https://wiki.m3u4u.com/en/Addendum#emby-pvrlive-kodisimpleiptv-tvheadends).

The problem is that they block the use of the service in Kodi among other services:

> Please Note, these services all have an inset bias to make an unwarranted number of download requests (epg) to the m3u4u server (in excess of 200 per minute). It is clear that many users of these services either do not know how or are not inclined to address the issue.

The solution is:

> An alternative to the script for Kodi+SimpleIPTV is to use [IPTV Merge add-on from MJH](https://www.matthuisman.nz/2019/02/iptv-merge-kodi-add-on.html) as your playlist and epg downloader which has a 24hr default.

**Comment by unkn0wn234** [comment]
> In kodi 19 there is an advanced setting to set this in iptvsimple settings. Try that.
> 
> EDIT: NVM, just read the end of your reply.

Hi I think I am facing the same problem here is my debug log its on a fresh install of Kodi 19 I tried the user agent method and the advanced settings xml

[kodi.old.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846378/kodi.old.log)
[kodi.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846377/kodi.log)

**Comment by unkn0wn234** [comment]
> > In kodi 19 there is an advanced setting to set this in iptvsimple settings. Try that.
> > EDIT: NVM, just read the end of your reply.
> 
> Hi I think I am facing the same problem here is my debug log its on a fresh install of Kodi 19 I tried the user agent method and the advanced settings xml
> 
> [kodi.old.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846378/kodi.old.log)
> [kodi.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846377/kodi.log)

Tried the IPTV Merge thing too same problem

[kodi.old.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846517/kodi.old.log)
[kodi.log](https://github.com/kodi-pvr/pvr.iptvsimple/files/5846518/kodi.log)

**Comment by phunkyfish** [comment]
@mathu204

There is something very wrong with your guide data: `Combined channel EPG with id '' now has display names: 'NETHERLANDS | FOX SPORTS 5 HD'`

Somehow, and I'm not sure sure how all your channel id's are empty. This should not be possible. 

Plus I don't see an attempted connection in your log. Did you try to play a channel?

**Comment by unkn0wn234** [comment]
> @mathu204
> 
> There is something very wrong with your guide data: `Combined channel EPG with id '' now has display names: 'NETHERLANDS | FOX SPORTS 5 HD'`
> 
> Somehow, and I'm not sure sure how all your channel id's are empty. This should not be possible.
> 
> Plus I don't see an attempted connection in your log. Did you try to play a channel?

Yes I’ve now gotten the channels working, titles and names are there but no EPg data, i opened the issue on the forum, I fixed my channels not loading from the m3u by turning off cache playlist

**Comment by unkn0wn234** [comment]
https://forum.kodi.tv/showthread.php?tid=359988

**Comment by phunkyfish** [comment]
Your channels have no id's:

```
<channel id="">
<display-name>NETHERLANDS | FOX SPORTS 5 HD</display-name>
<icon src="https://cdn.britannica.com/82/2982-004-ABEB3852/flag-prototype-Netherlands-countries-European-flags.jpg"/>
</channel>
```
Also I see many programme entries with no corresponding channels in the xmltv file.

TL;DR; your xmltv is not valid.

**Comment by unkn0wn234** [comment]
> Your channels have no id's:
> 
> ```
> <channel id="">
> <display-name>NETHERLANDS | FOX SPORTS 5 HD</display-name>
> <icon src="https://cdn.britannica.com/82/2982-004-ABEB3852/flag-prototype-Netherlands-countries-European-flags.jpg"/>
> </channel>
> ```
> 
> Also I see many programme entries with no corresponding channels in the xmltv file.
> 
> TL;DR; your xmltv is not valid.

Alright Ive takes it to my provider but the same Exact xml works on v18

**Comment by phunkyfish** [comment]
For v19 the validity of the xmltv format is validated.

But there are so many channels it’s almost impossible to see the wood from the trees.

If you can trim both the m3u8 and the xmltv file down to a single channel that works in v18 and not in v19 I can try to figure out a workaround.

**Comment by unkn0wn234** [comment]
> For v19 the validity of the xmltv format is validated.
> 
> But there are so many channels it’s almost impossible to see the wood from the trees.
> 
> If you can trim both the m3u8 and the xmltv file down to a single channel that works in v18 and not in v19 I can try to figure out a workaround.

working on getting it down to one file in both, would u need the links or just another debug log of me scanning the channels into kodi through iptv simple?

**Comment by unkn0wn234** [comment]
> > For v19 the validity of the xmltv format is validated.
> > But there are so many channels it’s almost impossible to see the wood from the trees.
> > If you can trim both the m3u8 and the xmltv file down to a single channel that works in v18 and not in v19 I can try to figure out a workaround.
> 
> working on getting it down to one file in both, would u need the links or just another debug log of me scanning the channels into kodi through iptv simple?

Also upon further checking through Ctrl+F that channel doesnt have an ID as you showed, but other channels do so I assume the EPG is just not complete but the channels that do have the ID are not being scanned in either none of them are, im trying to use m3u editor online to trim the m3u and xml so one sec

**Comment by phunkyfish** [comment]
If you do manage to find an example you can send it to me on gitter.im if you would rather not share it here.

**Comment by unkn0wn234** [comment]
> If you do manage to find an example you can send it to me on gitter.im if you would rather not share it here.



> If you do manage to find an example you can send it to me on gitter.im if you would rather not share it here.

Im fine sharing it but I messaged you because Im not sure I made the XML correctly I would appreciate some further help, and thank you for helpng me out already so far

**Comment by phunkyfish** [comment]
> > If you do manage to find an example you can send it to me on gitter.im if you would rather not share it here.
> 
> > If you do manage to find an example you can send it to me on gitter.im if you would rather not share it here.
> 
> Im fine sharing it but I messaged you because Im not sure I made the XML correctly I would appreciate some further help, and thank you for helpng me out already so far

Ok, I’ll respond on gitter

**Comment by phunkyfish** [comment]
Resolved,

**Comment by praveenrai2k-rgb** [comment]
Try downloading demo iptv Client and iptv simple client playlist will start loading.
